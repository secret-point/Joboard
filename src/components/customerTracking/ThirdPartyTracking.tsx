import React from "react";
import { Helmet } from "react-helmet";

// Added 2 new customerTracking places unique to the new URLs to track site traffic and completed applications
export const ThirdPartyTracking = () => {

  return (
    <Helmet>
      <script src="https://tracking.pandoiq.com/8950" async />
    </Helmet>
  );
};

export default ThirdPartyTracking;
