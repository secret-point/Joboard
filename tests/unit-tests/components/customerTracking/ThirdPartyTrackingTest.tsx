import React from "react";
import ThirdPartyTracking from "../../../../src/components/customerTracking/ThirdPartyTracking";
import { mount } from "enzyme";

const getItemSpy = jest.spyOn(window.localStorage, "getItem");

describe("ThirdPartyTracking", () => {

  beforeEach(() => {
    getItemSpy.mockReset();
  });

  it("should match snapshot on bb", () => {
    const url = "https://beta-us.devo.jobsatamazon.hvh.a2z.com/application/";
    Object.defineProperty(window, "location", {
      value: new URL(url)
    } );
    getItemSpy.mockReturnValue("nhe");
    const wrapper = mount(<ThirdPartyTracking />);
    expect(wrapper).toMatchSnapshot();
  });
});
