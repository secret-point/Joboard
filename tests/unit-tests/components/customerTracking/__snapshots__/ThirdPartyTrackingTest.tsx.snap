// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdPartyTracking should match snapshot on bb 1`] = `
<ThirdPartyTracking>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      script={
        Array [
          Object {
            "async": true,
            "src": "https://tracking.pandoiq.com/8950",
          },
        ]
      }
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        script={
          Array [
            Object {
              "async": true,
              "src": "https://tracking.pandoiq.com/8950",
            },
          ]
        }
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</ThirdPartyTracking>
`;
